src/pymor/reductors/neural_network.py

```diff
@@ -296,22 +298,121 @@ def build_basis(self):
 
                 # determine the coefficients of the full-order solutions in the reduced basis to obtain the
                 # training data; convert everything into tensors that are compatible with PyTorch
-                for mu, u in zip(self.training_set, U):
-                    mu_tensor = torch.DoubleTensor(mu.to_numpy())
-                    u_tensor = torch.DoubleTensor(reduced_basis.inner(u)[:,0])
-                    self.training_data.append((mu_tensor, u_tensor))
+                self.training_data = self._compute_samples(self.training_set, U, reduced_basis)
 
             # compute mean square loss
             mean_square_loss = (sum(U.norm2()) - sum(svals**2)) / len(U)
 
             return reduced_basis, mean_square_loss
 
+
+        def _compute_samples(self, parameters, solutions, reduced_basis):
+            """Transform parameters and corresponding solutions to tensors."""
+            samples = []
+            for mu, u in zip(parameters, solutions):
+                mu_tensor = torch.DoubleTensor(mu.to_numpy())
+                u_tensor = torch.DoubleTensor(reduced_basis.inner(u)[:,0])
+                samples.append((mu_tensor, u_tensor))
+            return samples
+
         def reconstruct(self, u):
             """Reconstruct high-dimensional vector from reduced vector `u`."""
             assert hasattr(self, 'reduced_basis')
             return self.reduced_basis.lincomb(u.to_numpy())
 
 
+    class NeuralNetworkInstationaryReductor(NeuralNetworkReductor):
+        """Reduced Basis reductor for instationary problems relying on
+        artificial neural networks.
+
+        This is a reductor that constructs a reduced basis using proper
+        orthogonal decomposition and trains a neural network that approximates
+        the mapping from parameter and time space to coefficients of the
+        full-order solution in the reduced basis.
+        The approach is described in [WHR19]_.
+
+        Parameters
+        ----------
+        fom
+            The full-order |Model| to reduce.
+        Nt
```

Comment:
I guess these have to be aligned with the snapshot data, so it would probably be better to derive them from `fom`?

Reply:
Probably, the approach also works with adaptive timesteppping to some extend. Unfortunately, we don't have any convention to get the corresponding time instances for a solution. (In pyMOR, we only have uniform time stepping in the first place. For external `Models`, `compute` could return these as an additional value.)